src/System.CommandLine/Suggestions/ISuggestionSource.cs

```diff
@@ -6,8 +6,17 @@
 
 namespace System.CommandLine.Suggestions
 {
+    /// <summary>
+    /// Defines the functionality to suggest values.
```

Comment:
Maybe something like "Provides suggestions for tab completion and example values for help", to give a sense of when we might suggest values.